chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/watcher/outgoing/ChaosMonkeyRestTemplateWatcher.java

```diff
@@ -0,0 +1,103 @@
+package de.codecentric.spring.boot.chaos.monkey.watcher.outgoing;
+
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosMonkeyRequestScope;
+import de.codecentric.spring.boot.chaos.monkey.component.ChaosTarget;
+import de.codecentric.spring.boot.chaos.monkey.configuration.AssaultProperties;
+import de.codecentric.spring.boot.chaos.monkey.configuration.WatcherProperties;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Random;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.AbstractClientHttpResponse;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+/** @author Marcel Becker */
+public class ChaosMonkeyRestTemplateWatcher implements ClientHttpRequestInterceptor {
+
+  private final WatcherProperties watcherProperties;
+  private final ChaosMonkeyRequestScope chaosMonkeyRequestScope;
+  private final AssaultProperties assaultProperties;
+
+  public ChaosMonkeyRestTemplateWatcher(
+      final ChaosMonkeyRequestScope chaosMonkeyRequestScope,
+      final WatcherProperties watcherProperties,
+      AssaultProperties assaultProperties) {
+    this.chaosMonkeyRequestScope = chaosMonkeyRequestScope;
+    this.watcherProperties = watcherProperties;
+    this.assaultProperties = assaultProperties;
+  }
+
+  @Override
+  public ClientHttpResponse intercept(
+      final HttpRequest httpRequest,
+      byte[] bytes,
+      ClientHttpRequestExecution clientHttpRequestExecution)
+      throws IOException {
+    ClientHttpResponse response;
+    response = clientHttpRequestExecution.execute(httpRequest, bytes);
+    if (watcherProperties.isRestTemplate()) {
+      try {
+        chaosMonkeyRequestScope.callChaosMonkey(
+            ChaosTarget.REST_TEMPLATE, this.getClass().getSimpleName());
```

Comment:
The class name won't change in a meaningful way. I would recommend naming the toggle based on the host. That way users can target the chaos against specific dependencies.

Reply:
I agree, I will change it to  `getURI().toUrl().toString()`